src/main/java/cn/nukkit/entity/Entity.java

```diff
@@ -1093,7 +1094,43 @@ public boolean attack(EntityDamageEvent source) {
             this.setAbsorption(Math.max(0, this.getAbsorption() + source.getDamage(EntityDamageEvent.DamageModifier.ABSORPTION)));
         }
         setLastDamageCause(source);
-        setHealth(getHealth() - source.getFinalDamage());
+
+        float newHealth = getHealth() - source.getFinalDamage();
+        if (newHealth < 1 && this instanceof Player) {
```

Comment:
Why not to implement this in the Player class then?

Reply:
That would have required the `attack` method to be refactored and/or duplicated code